### PR TITLE
Group method + Travis test

### PR DESCRIPTION
**Pull request by budnieswski**
I implemented a **group** nested method, it works in a way that does not need to create multiple instances of the Router (as if it were the standard Nesting Routers), it just concatenates the group prefixes and middlewares, until the moment of adding a real route.

What was done:
 - Lib of Javascript
 - types for Typescript
 - Mocha tests
 - Fixed the build tests
 - [Build in Travis CI](https://travis-ci.org/github/budnieswski/restify-router)

Below are some examples of implementation:
* [Javascript] Basic usage:
https://gist.github.com/budnieswski/c52c94019d758c9497e93de88a47792a

* [Javascript] Nested middleware usage:
https://gist.github.com/budnieswski/009637a3fb0c2a5f95b9261f5d6fdffb

* [Typescript] Basic usage:
https://gist.github.com/budnieswski/9b418f5cd961ddb01fb9a3c5391d3f0d

* [Typescript] Separated route file from server:
https://gist.github.com/budnieswski/b15eb44468c30b873497b8823a437464

### CODE REVIEW

**Comment by ukayani** [comment]
Hi @budnieswski ,

Thanks for your contribution!

I'm wondering what the motivation is behind this change since you can accomplish this nesting functionality with nested routers. Is it purely so you dont have to define the extra instances?

**Comment by budnieswski** [comment]
Hi @ukayani ,

The main reason was the manipulation of middlewares, and the organization of routes / files.

The separation of routes into several files it's repetitive and complicate middlewares usage. On the other hand, if only one file was used, the use of several instances pollutes the code, getting a little "confusing". That was where the group's idea came from, visually pleasing and easy to handle!

I am open to criticism and suggestions = P

File: gulpfile.js [hunk 1]
```diff
@@ -50,4 +50,5 @@ gulp.task('lint', function () {
     .pipe(jshint.reporter('fail'));
 });
 
-gulp.task('default', ['test', 'lint', 'style']);
+// gulp.task('default', ['test', 'lint', 'style']); // Gulp v3.x
+gulp.task('default', gulp.parallel('test', 'lint', 'style')); // Gulp v4.x
```
Comment (ukayani) [inline gulpfile.js:54]: thanks for updating

File: test/router.spec.js [hunk 2]
```diff
@@ -592,6 +592,216 @@ describe('Restify Router', function () {
...
+          done();
+        });
+
+    });
+
+    it('Should fail if invalid path type is provided for router', function () {
+
+      function fail() {
+        var router = new Router();
+        var nested = new Router();
```
Comment (ukayani) [inline test/router.spec.js:662]: is this used?

File: gulpfile.js (hunk 1 above)
Comment (ukayani) [reply gulpfile.js:54]: I decided to remove gulp from the build process and opted to use the scripts functionality of npm instead.

File: test/router.spec.js (hunk 2 above)
Comment (budnieswski) [reply test/router.spec.js]: This variable has been forgotten, it is not used =P

I updated the fork with the master and removed this variable.

**Comment by ukayani** [comment]
@budnieswski thanks for the contribution. I know i've merged this PR already but if you get some time, could you please update the README with the new syntax? Thanks

**Comment by budnieswski** [comment]
> @budnieswski thanks for the contribution. I know i've merged this PR already but if you get some time, could you please update the README with the new syntax? Thanks

Sure, I'll do that  between today and tomorrow!

**Pull request merged**